### PR TITLE
fix_torque_maps_api_template

### LINKED ISSUE #9017: Torque doesn't work with shared datasets
STR

Disable builder in order to try with the Editor (I haven't been able to test this in the Builder because of a bug but I think that the problem is exactly the same).

1. Go to Dashboard -> Datasets -> Shared with you.
2. Select a private one and click "Create map".
3. Open the map in an incognito window. It works (this helps to see that problem is at torque).
4. Set Torque style in the wizard. There's a GET to the tiler that fails with this error:

```
{type: "unknown", message: "TorqueRenderer: permission denied for relation twitter_t3chfest_reduced",…}
```

That request is done to `https://TABLE_OWNER_USERNAME.carto.com/api/v1/map?config=...`, which I think that is wrong, as it should be done to the subdomain of the owner of the visualization.

5. If you open the embed the request to the tiler fails with `"Template 'tpl_9b364c98_4e90_11e6_937a_0e3ebc282e83' of user 'TABLE_OWNER_USERNAME' not found"`, because of the same reason.

6. If you reload the editor the Torque animation works, as the request is now done with the right user subdomain (the owner of the visualization). Nevertheless the embed seems broken again.

It looks like Torque is creating the url with the username that it finds at `layer.options['user_name']`. That's wrong, as that user is not the visualization owner but the `layer.options['table_name']` owner.

This has appeared while testing the fix for #8974.

cc @xavijam @alonsogarciapablo @fdansv @javisantana

### PR DESCRIPTION
**Pull request by juanignaciosl**
This fixes #9017.

### CODE REVIEW

File: app/models/layer/presenter.rb [hunk 1]
```diff
@@ -141,7 +141,7 @@ def as_torque
           legend:     layer.legend,
           options:    {
             stat_tag:           options.fetch(:visualization_id),
-            maps_api_template: ApplicationHelper.maps_api_template(api_templates_type),
+            maps_api_template: fix_torque_maps_api_template(ApplicationHelper.maps_api_template(api_templates_type)),
```
Comment (javisantana) [inline app/models/layer/presenter.rb]: just to be totally sure, this is only for torque layers, right?

Comment (juanignaciosl) [inline app/models/layer/presenter.rb]: Yep, this new patch method is used only at `as_torque` (and I've included `torque` in the name to make that crystal clear.

What do you think about the patch?

Comment (javisantana) [inline app/models/layer/presenter.rb]: sounds good, just wanted to double check

**Comment by javisantana** [comment]
:es:

**Comment by juanignaciosl** [comment]
@javisantana ok!

@javitonino CR this, please. I'll probably merge it into the `8974-Create_map_with_shared_dataset_doesnt_work` and test it altogether, as it should fix all scenarios.

**Comment by javitonino** [comment]
LGTM :+1:

**Pull request merged**